### PR TITLE
Change the way ActionDispatcher is found.

### PR DESCRIPTION
**Pull request by gspencergoog**
## Description

This changes the way `ActionDispatcher`s are found by the `Actions` widget, so that by default it will look for dispatchers of the parent Actions widgets instead of just creating a default `ActionDispatcher`.  This allows overriding of the ActionDispatcher at the top level: before, the custom action dispatcher would only be invoked if explicitly set on all the Actions widgets.

This is not a breaking change because there was a default value to the `dispatcher` parameter before that performed this function, and not specifying the dispatcher anywhere will still result in a default dispatcher being created.

## Tests

- Added a test to make sure that the parent's custom dispatcher would be found.

## Checklist

Before you create this PR confirm that it meets all requirements listed below by checking the relevant checkboxes (`[x]`). This will ensure a smooth and quick review process.

- [X] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [X] I signed the [CLA].
- [X] I read and followed the [Flutter Style Guide], including [Features we expect every widget to implement].
- [X] I updated/added relevant documentation (doc comments with `///`).
- [X] All existing and new tests are passing.
- [X] The analyzer (`flutter analyze --flutter-repo`) does not report any problems on my PR.
- [X] I am willing to follow-up on review comments in a timely manner.
- [X] I checked all the boxes!

## Breaking Change

Does your PR require Flutter developers to manually update their apps to accommodate your change?

- [X] No, this is *not* a breaking change.

[issue database]: https://github.com/flutter/flutter/issues
[Contributor Guide]: https://github.com/flutter/flutter/wiki/Tree-hygiene#overview
[Test Coverage]: https://github.com/flutter/flutter/wiki/Test-coverage-for-package%3Aflutter
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[Features we expect every widget to implement]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo#features-we-expect-every-widget-to-implement
[CLA]: https://cla.developers.google.com/
[Handling breaking changes]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes

### CODE REVIEW

File: packages/flutter/test/widgets/actions_test.dart [hunk 1]
```diff
@@ -254,14 +295,18 @@ void main() {
...
+        child: Container(),
+      ).debugFillProperties(builder);
 
       final List<String> description = builder.properties
-        .where((DiagnosticsNode node) {
-          return !node.isFiltered(DiagnosticLevel.info);
-        })
-        .map((DiagnosticsNode node) => node.toString())
-        .toList();
+          .where((DiagnosticsNode node) {
```
Comment (darrenaustin) [inline packages/flutter/test/widgets/actions_test.dart]: Why the change in indentation here? I thought it was supposed to be two spaces for this kind of expression.

File: packages/flutter/lib/src/widgets/actions.dart [hunk 2]
```diff
@@ -243,18 +289,24 @@ class Actions extends InheritedWidget {
   }) {
     assert(context != null);
     assert(intent != null);
-    Actions actions;
+    Element actionsElement;
     Action action;
+
+    if (identical(intent, Intent.doNothing)) {
```
Comment (darrenaustin) [inline packages/flutter/lib/src/widgets/actions.dart]: It would be good to have a test for this (unless I missed it below).

**Review by darrenaustin — APPROVED** [review]
LGTM.

Just some minor comments.

**Comment by gspencergoog** [comment]
Crap.  I didn't mean to commit this yet.  I'll submit another PR that addresses your comments @darrenaustin
